testsuite/tests/typing-jkind-bounds/no-infer-across-modules/define_with_kinds.ml

```diff
@@ -0,0 +1,27 @@
+type 'a my_list =
+  | Nil
+  | Cons of 'a * 'a my_list
+
+type 'a my_ref = { mutable contents : 'a }
+
+type 'a rec1 =
+  | Foo1 of 'a rec2 * int
+  | Bar1 of 'a
+
+and 'a rec2 =
+  | Foo2 of 'a rec1 * int
+  | Bar2 of 'a
+
+type ('a : value & value) unboxed_rec = #{ x : 'a ; y : string }
```

Comment:
I object to this name since the previous type uses `rec` to mean "recursive" and here it means "record"